src/huggingface_hub/repository.py

```diff
@@ -547,6 +555,30 @@ def current_branch(self) -> str:
 
         return result
 
+    @property
+    @_deprecate_method(
+        version="0.12",
+        message="Attribute `repo_type` is not a reliable information.",
```

Comment:
As a user, I'm not sure I'd understand what's wrong with the `repo_type` and why it's not reliable information. Personally I'd find it cleaner to say that it's deprecated alongside the `clone_from` and what to use instead

```suggestion
        message="Attribute `repo_type` is not a reliable information.",
```

Reply:
```suggestion
        message="`repo_type` is only used in a deprecated use case of `clone_from`.",
```